### PR TITLE
Ensure service install directories are created before writing file

### PR DESCRIPTION
**Pull request by heckler1**
Service install fails on a fresh macOS install

### CODE REVIEW

File: cmd/cloudflared/service_template.go [hunk 1]
```diff
@@ -48,10 +49,17 @@ func (st *ServiceTemplate) Generate(args *ServiceTemplateArgs) error {
...
-	fileMode := os.FileMode(0644)
+	fileMode := os.FileMode(0o644)
 	if st.FileMode != 0 {
 		fileMode = st.FileMode
 	}
+
+	plistFolder := path.Dir(resolvedPath)
+	err = os.MkdirAll(plistFolder, 0o755)
+	if err != nil {
+		return fmt.Errorf("error creating %s: %v", plistFolder, err)
```
Comment (sudarshan-reddy) [inline cmd/cloudflared/service_template.go:60]: Could this also error if the directories exist? Should we ignore `IsExists` error?

Comment (heckler1) [reply cmd/cloudflared/service_template.go:60]: No, MkdirAll returns `nil` if the directory exists, similar to `mkdir -p`. 

https://pkg.go.dev/os#MkdirAll

**Comment by heckler1** [comment]
Not sure why the checks are failing, I don't see anything in the logs that looks related to this change. Could we try a re-run?

**Review by sudarshan-reddy — APPROVED** [review]

**Comment by heckler1** [comment]
Bumping, would like to get this merged if possible

**Comment by DevinCarr** [comment]
Can you rebase against the latest master? I'm unable to re-run the current builds that are failing.

**Review by DevinCarr — APPROVED** [review]

**Review by ipostelnik — APPROVED** [review]

**Review by chungthuang — APPROVED** [review]

**Comment by heckler1** [comment]
I think this is ready to merge if an authorized member can skip the flaky CI

**Review by sssilver — APPROVED** [review]

**Pull request merged**